src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacKeyFieldDeclaration/KeyFieldDeclarationFix.cs

```diff
@@ -0,0 +1,134 @@
+																	cToken => RemoveKeysFromFieldsAsync(context.Document, 
+																										context,
+																										cToken, 
+																										diagnostic, 
+																										editIdentityAttribute: true),
+																	equivalenceKey: codeActionBoundKeysName);
+
+				string codeActionRemoveIdentityColumnName = nameof(Resources.PX1055Fix3).GetLocalized().ToString();
+				CodeAction codeActionRemoveIdentityColumn = CodeAction.Create(codeActionRemoveIdentityColumnName,
+																				cToken => RemoveKeysFromFieldsAsync(context.Document, 
+																													context, 
+																													cToken, 
+																													diagnostic,
+																													removeIdentityAttribute: true),
+																				equivalenceKey: codeActionRemoveIdentityColumnName);
+
+				context.RegisterCodeFix(codeActionIdentityKey, context.Diagnostics);
+				context.RegisterCodeFix(codeActionBoundKeys, context.Diagnostics);
+				context.RegisterCodeFix(codeActionRemoveIdentityColumn, context.Diagnostics);
+			}, context.CancellationToken);
+			
+		}
+
+		private async Task<Document> RemoveKeysFromFieldsAsync(Document document,CodeFixContext context, CancellationToken cToken, Diagnostic diagnostic, bool editIdentityAttribute = false, bool removeIdentityAttribute = false)
+		{
+
+			SemanticModel semanticModel = await document.GetSemanticModelAsync(cToken).ConfigureAwait(false);
+			PXContext pxContext = new PXContext(semanticModel.Compilation);
+			AttributeInformation attributeInformation = new AttributeInformation(pxContext);
+
+			cToken.ThrowIfCancellationRequested();
+
+			Location[] attributeLocations = diagnostic.AdditionalLocations.ToArray();
+			var identityAttributeType = pxContext.FieldAttributes.PXDBIdentityAttribute;
+			var longIdentityAttributeType = pxContext.FieldAttributes.PXDBLongIdentityAttribute;
+
+			if (attributeLocations == null)
+				return document;
+
+			SyntaxNode root = await document.GetSyntaxRootAsync(cToken).ConfigureAwait(false);
+
+			Document tempDocument = document;
+
+			List<SyntaxNode> deletedNodes = new List<SyntaxNode>();
+
+			foreach (var attributeLocation in attributeLocations)
+			{
+				AttributeSyntax attributeNode = root?.FindNode(attributeLocation.SourceSpan) as AttributeSyntax;
+
+				if (attributeNode == null)
+					return document;
+
+				ITypeSymbol attributeType = semanticModel.GetTypeInfo(attributeNode, cToken).Type;
+
+				if (attributeType == null)
+					return document;
+
+				bool isIdentityAttribute = attributeInformation.IsAttributeDerivedFromClass(attributeType, identityAttributeType) || 
+										   attributeInformation.IsAttributeDerivedFromClass(attributeType, longIdentityAttributeType);
+
+				if (removeIdentityAttribute == false && !isIdentityAttribute ^ editIdentityAttribute)
+				{
+					var deletedNode = attributeNode.ArgumentList.Arguments.Where(a => a.NameEquals?.Name.Identifier.ValueText.Equals(IsKey)??false && 
```

Comment:
spaces

Reply:
The predicate in Where method  is hard to understand and unobvious. It should be refactored, maybe placed in a separate method.